src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1868,16 +1868,69 @@ private static Expression BindParameterFromValue(ParameterInfo parameter, Expres
         var parameterNameConstant = Expression.Constant(parameter.Name);
         var sourceConstant = Expression.Constant(source);
 
+        if (source == "header" && (parameter.ParameterType.IsArray || typeof(IEnumerable<string>).IsAssignableFrom(parameter.ParameterType)))
+        {
+            var stringValuesExpr = Expression.Convert(valueExpression, typeof(StringValues));
+            var toStringArrayMethod = typeof(StringValues).GetMethod(nameof(StringValues.ToArray));
+            if (toStringArrayMethod == null)
+            {
+                throw new InvalidOperationException("The method 'ToArray' could not be found on 'StringValues'.");
+            }
+            var headerValuesArrayExpr = Expression.Call(stringValuesExpr, toStringArrayMethod);
+
+            var splitAndTrimMethod = typeof(RequestDelegateFactory).GetMethod(nameof(SplitAndTrim), System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic);
+            if (splitAndTrimMethod == null)
+            {
+                throw new InvalidOperationException("The method 'SplitAndTrim' could not be found on 'RequestDelegateFactory'.");
+            }
+            var splitAndTrimExpr = Expression.Call(splitAndTrimMethod, headerValuesArrayExpr);
+
+            var boundValueExpr = Expression.Convert(splitAndTrimExpr, parameter.ParameterType);
+
+            if (!isOptional)
```

Comment:
Looks like this is all duplicate code? Can you just set `valueExpression = boundValueExpr;` and remove all the additional code below this?